flask_project/campaign_manager/test/test_models.py

```diff
@@ -1,573 +1,296 @@
 import os
-from unittest import TestCase
+import ast
+import datetime
+
+from flask_testing import TestCase
+from flask import Flask
 from campaign_manager.models.models import *
+from geoalchemy2.shape import from_shape
+from shapely.geometry import asShape
+from sqlalchemy import select, func
 from sqlalchemy.orm import sessionmaker
+from campaign_manager.sqlalchemy_session import engine, session, Base
+
+
+""" Geometry object for mock campaign """
+geom = {
+        "type": "Polygon",
+        "coordinates": [
+                [
+                    [
+                        -22.7271314710379,
+                        23.9871672761574
+                    ],
+                    [
+                        -22.7271314710379,
+                        23.9879122455024
+                    ],
+                    [
+                        -22.7264448255301,
+                        23.9879122455024
+                    ],
+                    [
+                        -22.7264448255301,
+                        23.9871672761574
+                    ],
+                    [
+                        -22.7271314710379,
+                        23.9871672761574
+                    ]
+                ]
+            ]
+        }
 
 
-def db_setUp():
-    db_location = os.environ['TESTDATABASE_URL']
-    engine = create_engine(db_location, echo=True)
-    return engine
-
+class TestBase(TestCase):
+    def create_app(self):
+        app = Flask(__name__)
+        return app
 
-# Destroy the mock test Database
-def db_destroy(engine):
-    for tbl in reversed(meta.sorted_tables):
-        engine.execute(tbl.delete())
 
+def setUp():
```

Comment:
So before each test, you create an user, a campaign, a feature_type, a feature_template, a taskBoundary (should be task_boundary btw), a team and a function. That's too much and might slow the test suite.
You should only create objects your tests need and destroy them in the tearDown method.

Reply:
Take a look at [FactoryBoy](http://factoryboy.readthedocs.io/en/latest/)